libcxx/include/typeinfo

```diff
@@ -276,7 +276,19 @@ struct __type_info_implementations {
           __impl;
 };
 
-class _LIBCPP_EXPORTED_FROM_ABI type_info {
+#    if defined(__arm64__) && __has_cpp_attribute(clang::ptrauth_vtable_pointer)
```

Comment:
This seems like configuration that might be better suited for a config header?

type_info is weird magic though, so I understand if you disagree.

Reply:
Separately, to support ELF platforms that don't define `__arm64__` we ended up rewriting the feature check as:
```
+#if __has_feature(ptrauth_calls) &&                          \
+    (__has_feature(ptrauth_vtable_address_discrimination) || \
+     __has_feature(ptrauth_vtable_type_discrimination)) &&   \
+    __has_cpp_attribute(clang::ptrauth_vtable_pointer)
```